tests/by-util/test_groups.rs

```diff
@@ -1,41 +1,53 @@
 use crate::common::util::*;
 
 #[test]
+#[cfg(any(target_vendor = "apple", target_os = "linux"))]
 fn test_groups() {
-    let result = new_ucmd!().run();
-    println!("result.stdout = {}", result.stdout_str());
-    println!("result.stderr = {}", result.stderr_str());
-    if is_ci() && result.stdout_str().trim().is_empty() {
-        // In the CI, some server are failing to return the group.
-        // As seems to be a configuration issue, ignoring it
-        return;
+    if !is_ci() {
+        new_ucmd!().succeeds().stdout_is(expected_result(&[]));
+    } else {
+        // TODO: investigate how this could be tested in CI
+        // stderr = groups: cannot find name for group ID 116
+        println!("test skipped:");
     }
-    result.success();
-    assert!(!result.stdout_str().trim().is_empty());
 }
 
 #[test]
-fn test_groups_arg() {
-    // get the username with the "id -un" command
-    let result = TestScenario::new("id").ucmd_keepenv().arg("-un").run();
-    println!("result.stdout = {}", result.stdout_str());
-    println!("result.stderr = {}", result.stderr_str());
-    let s1 = String::from(result.stdout_str().trim());
-    if is_ci() && s1.parse::<f64>().is_ok() {
-        // In the CI, some server are failing to return id -un.
-        // So, if we are getting a uid, just skip this test
-        // As seems to be a configuration issue, ignoring it
+#[cfg(any(target_os = "linux"))]
+#[ignore = "fixme: 'groups USERNAME' needs more debugging"]
+fn test_groups_username() {
+    let scene = TestScenario::new(util_name!());
+    let whoami_result = scene.cmd("whoami").run();
```

Comment:
Any possibility to get this without the `whoami` utility? Via libc? Although I guess that could be considered a little bit circular, as we're also using `libc` to get it in the first place. Maybe it would also work on the CI then?

Reply:
see my other comment